#! /bin/sh /usr/share/dpatch/dpatch-run
## 00_empty_ip.dpatch by  <debian@onerussian.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Filter out empty IPs for ignoreip

@DPATCH@

diff -x '*~' -Naur fail2ban-0.6.1.pre/fail2ban.py fail2ban-0.6.1.post/fail2ban.py
--- fail2ban-0.6.1.pre/fail2ban.py	2006-03-19 00:20:44.000000000 -0500
+++ fail2ban-0.6.1.post/fail2ban.py	2006-07-03 21:56:10.000000000 -0400
@@ -356,7 +356,9 @@
 					"ONLY DISPLAYED IN THE LOG MESSAGES")
 	
 	# Ignores IP list
-	ignoreIPList = conf["ignoreip"].split(' ')
+	# and filter out empty entries. Otherwise
+	#  WARNING:  is not a valid IP address
+	ignoreIPList = filter(None, conf["ignoreip"].split(' '))
 
 	# Checks for root user. This is necessary because log files
 	# are owned by root and firewall needs root access.
